lambdas/classifier/app.py

```diff
@@ -0,0 +1,99 @@
+import itertools
+import json
+import re
+from datetime import datetime
+
+import boto3
+
+MANIFEST_PREFIX_V1 = ".quilt/packages/"
+POINTER_PREFIX_V1 = ".quilt/named_packages/"
+
+
+event_bridge = boto3.client('events')
+s3 = boto3.client('s3')
+
+
+class EventsQueue:
+    MAX_SIZE = 10
+
+    def __init__(self):
+        self._events = []
+
+    def append(self, event):
+        self._events.append(event)
+        if len(self) >= self.MAX_SIZE:
+            self._flush()
+
+    def _flush(self):
+        events = self._events
+        self._events = []
+        event_bridge.put_events(Entries=events)
+        print(events)
+
+    def flush(self):
+        if self:
+            self._flush()
+
+    def __len__(self):
+        return len(self._events)
+
+    def __bool__(self):
+        return bool(self._events)
+
+
+PKG_POINTER_REGEX = re.compile(r"\.quilt/named_packages/([\w-]+/[\w-]+)/(.{10})")
+
+
+def pkg_created_event(s3_event):
+    print(s3_event)
+    if not s3_event['eventName'].startswith('ObjectCreated:'):
+        return
+    s3_event_obj = s3_event['s3']
+    obj = s3_event_obj['object']
+    key = obj['key']
+    match = PKG_POINTER_REGEX.fullmatch(key)
+    if not match:
+        return
+    pkg_name, pointer_name = match.groups()
+    if not '1451631600' <= pointer_name <= '1767250800':
+        return
+    bucket_obj = s3_event_obj['bucket']
+    bucket = bucket_obj['name']
+    etag = obj['eTag']
+    version_id = obj['versionId']
+    pkg_hash = s3.get_object(Bucket=bucket, Key=key, IfMatch=etag, VersionId=version_id)['Body'].read().decode()
+    manifest_key = MANIFEST_PREFIX_V1 + pkg_hash
+    resp = s3.head_object(Bucket=bucket, Key=manifest_key)
+    return {
+        'Time': datetime.fromisoformat(s3_event['eventTime'].rstrip('Z')),
+        'Source': 'quiltdata.pkg',
```

Comment:
com.quiltdata? eventbridge.quiltdata?

Reply:
Probably we should consider https://docs.aws.amazon.com/eventbridge/latest/userguide/aws-events.html and look through https://docs.aws.amazon.com/eventbridge/latest/userguide/event-types.html.

Reply:
Totally agree. In particular glue catalog events are a good pattern for us:

```
{
    "version": "0",
    "id": "2617428d-715f-edef-70b8-d210da0317a0",
    "detail-type": "Glue Data Catalog Table State Change",
    "source": "aws.glue",
    "account": "123456789012",
    "time": "2019-01-16T18:16:01Z",
    "region": "eu-west-1",
    "resources": [
        "arn:aws:glue:eu-west-1:123456789012:table/d1/t1"
    ],
    "detail": {
        "databaseName": "d1",
        "changedPartitions": [
            "[C.pdf, dir3]",
            "[D.doc, dir4]"
        ],
        "typeOfChange": "BatchCreatePartition",
        "tableName": "t1"
    }
}
```

Reply:
> Actually `eventBridge:quilt`.

https://docs.aws.amazon.com/eventbridge/latest/userguide/aws-events.html

> source

> Identifies the service that sourced the event. All events sourced from within AWS begin with "aws." Customer-generated events can have any value here, as long as it doesn't begin with "aws." **We recommend the use of Java package-name style reverse domain-name strings.**

i.e. it would be like `com.quiltdata` or `com.quiltdata.XXX`.

Reply:
I looked through different event types and it looks like they use different property naming conventions: [some](https://docs.aws.amazon.com/eventbridge/latest/userguide/event-types.html#acd-event-types) use camel case and [others](https://docs.aws.amazon.com/eventbridge/latest/userguide/event-types.html#codepipeline-event-type) use hyphens. There are some recommendations to use camel case ([1](https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format), [2](https://jsonapi.org/recommendations/#naming)), so I'm planning to stick to it.